stuffer/s2n_stuffer_network_order.c

```diff
@@ -190,6 +189,9 @@ int s2n_stuffer_write_reservation(struct s2n_stuffer_reservation reservation, ui
 
 int s2n_stuffer_write_vector_size(struct s2n_stuffer_reservation reservation)
```

Comment:
I was surprised to see that these functions took a reservation by value and not by reference (which is how the other reservation functions behave).

Is there a reason these behave differently? Do we know the impact of changing these functions to behave like the others?

Reply:
@camshaft @danielsn any thoughts? The only reason I can think of is the one Ryan pointed in the next [comment](https://github.com/awslabs/s2n/pull/2059#discussion_r456464834).

Reply:
I'd think passing by reference is the more appropriate approach, especially with it being larger than `size_t`.

Reply:
I modified the functions to take a reservation by reference.